caching-big-repositories_5bf03fb92c7d3a31944e139d.md

```diff
@@ -0,0 +1,105 @@
+
+* [Overview](#overview)
+* [The logic behind the recipe](#the-logic-behind-the-recipe)
+* [An example project](#an-example-project)
+* [See Also](#see-also)
+
+## Overview
+
+Imagine that you have one or multiple large files on a GitHub repository and
+you want to make the downloading of them as fast as possible.
+
+Note that for this recipe to work, the files of the GitHub repository should
+remain intact through the lifetime of the Semaphore project.
+
+## The logic behind the recipe
+
+The example project that follows is getting all the files of the GitHub
+repository in the first job using `checkout` and then is storing them into the
+Semaphore 2.0 Cache server using `cache store`. The remaining jobs of the
```

Comment:
I think that this recipe should do more than this.

- Keep the repo in cache throughout workflows so that even the first block doesn't need to do `git clone`;
- Pull the latest changes in requested branch only.
- This all assumes cache hit, but there should be a fallback in the first block in case of cache miss.

Reply:
I would suggest that we keep that recipe _simple_ and maybe create a new recipe with a more complex scenario.